### ISSUE TITLE
FP: Foreach variable with referenced and then unsetted variable still existing

### ISSUE THREAD

**Issue by CZechBoY** [kalessil/phpinspectionsea#329]
```php
foreach ($abc as &$a) {
    //code
}
unset($a);
foreach ($abc as $a) { // Probable bug: variable should be renamed to prevent writing into already existing reference.
    // code
}
```

**Comment by rentalhost** [comment]
While I think that it should be fixed, I too think that this code should be refactored.

**Comment by CZechBoY** [comment]
I can rename the variable in the second foreach of course. Or you mean something different?

**Comment by rentalhost** [comment]
Yes, rename is an option. But anyway it should be valid, because the unset() should "clear" the "used variable" status.

**Comment by rentalhost** [comment]
Moving to 3.0.+ because we need do some refactoring on this file and UTs to make sure do not break nothing (ping @kalessil).

**Comment by kalessil** [comment]
Sorry, but I promised to keep fixing bugs as prio 1 =)

**Comment by rentalhost** [comment]
No problem. I just worry about there is no UTs for this inspection currently. I tried to work on it but I was worried about breaking something haha

**Comment by kalessil** [comment]
I'll take care about

**Comment by kalessil** [comment]
For myself: workflow re-mastering in needed

- Simplify workflow by excluding in-loop usages (get group statements, search, exclude);

- Support following:
```php
foreach ([] as $i => &v) {
    ...
    unset($whatever[$i], $v);
}
```
